### DISCUSSION TITLE
KICS 1.6.x - Flags Deprecation

### DISCUSSION THREAD

**Discussion by cx-nuno-oliveira** [Checkmarx/kics#5751] [General]
Hi everybody, 

We are planning to deprecate some flags in KICS, during KICS 1.6.x releases.
The motivation: continuously keep "how to use KICS" documentation easy to read and follow; continuously keep KICS maintenance and improvements easy and accessible to the community.  

We would like to have your opinion and feedback on the deprecation of the following flags:

- ``--minimal-ui``
- ``--no-progress``
- ``--preview-lines``
- ``--no-color``
- ``--input-data``

Thanks

**Comment by lior-orca** [comment]
@nunoocx can you explain more about the reasoning behind the removal of `--preview-lines` flag?

**Reply by cx-nuno-oliveira** [reply]
Hi @lior-orca, the ``--preview-lines`` flag is used to increase/decrease the number of lines of the code exposed in the CLI output, associated to each result. At the moment we show 3 lines (the line of the issue, one above and one below) and the idea is to remain fixed at 3 lines.

While this view is helpful for an initial visualization, most of the cases it is necessary to go to the full code and understand the result for correction. Since presenting the full code is not practical, we believe that users end up going directly to the original files.
Therefore, we see this flag not that helpful.

This is just a proposal for deprecations, let us know if you think it should remain. Tell us your usage scenarios. :)

**Reply by lior-orca** [reply]
@nunoocx I think having the flag do require a  lot of "code maintenance" and can help in various use cases (e.g. decrease it to 1 preview line, to reduce the noise)

**Reply by cx-nuno-oliveira** [reply]
Thanks for the feedback @lior-orca. We will keep it active, then :) 
About the other, do you have any claim to keep them, or can we proceed with deprecation?
